test/locationscale.jl

```diff
@@ -173,4 +173,13 @@ end
         test_location_scale_discretenonparametric(_rng, -1//4, 1//3, (-10):(-1), probs)
         test_location_scale_discretenonparametric(_rng, 6//5, 3//2, 15:24, probs)
     end
+
+    @test_nowarn ad_norm = AffineDistribution(1.0, 1, Normal())
+    @test ad_norm isa AffineDistribution{Float64, Continuous, Normal{Float64}}
```

Comment:
As I assumed, `@test_nowarn` does not cover deprecation warnings. E.g., check the following example (have to start Julia with `julia --depwarn=yes`):
```julia
julia> using Test

julia> f(x, y) = x + y
f (generic function with 1 method)

julia> Base.@deprecate g(x, y) f(x, y)
g (generic function with 1 method)

julia> function h(x, y)
           print(stderr, "`h(x, y)` is deprecated, use `f(x, y)` instead.")
           return f(x, y)
       end
h (generic function with 1 method)

julia> @test_nowarn f(1, 2)
3

julia> @test_nowarn g(1, 2)
┌ Warning: g(x, y) is deprecated, use f(x, y) instead.
│   caller = #14 at Test.jl:801 [inlined]
└ @ Core /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/Test/src/Test.jl:801
3

julia> @test_nowarn h(1, 2)
`h(x, y)` is deprecated, use `f(x, y)` instead.Test Failed at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/Test/src/Test.jl:806
  Expression: isempty(stderr_content)
   Evaluated: isempty("`h(x, y)` is deprecated, use `f(x, y)` instead.")
ERROR: There was an error during testing
```
I.e., `@test_nowarn` passes also for the deprecated method `g` since it only checks `stderr`.

Instead one has to use `@test_logs` here:
```julia
julia> @test_logs f(1, 2)
3

julia> @test_logs g(1, 2)
Log Test Failed at REPL[19]:1
  Expression: g(1, 2)
  Log Pattern:
  Captured Logs:
    Test.LogRecord(Warn, "`g(x, y)` is deprecated, use `f(x, y)` instead.", Main, :depwarn, (Ptr{Nothing} @0x00007f37f8121ba9, :g), "./none", 0, Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:caller, :maxlog), Tuple{Base.Stac
kTraces.StackFrame, Int64}}}(:caller => (::var"#33#34")() at none:0, :maxlog => 1))

ERROR: There was an error during testing

julia> @test_logs h(1, 2)
`h(x, y)` is deprecated, use `f(x, y)` instead.3
```

Thus we should just check:
```suggestion
    @test (@test_logs Distributions.AffineDistribution(1.0, 1, Normal())) isa Distributions.AffineDistribution{Float64, Continuous, Normal{Float64}}
```
or just 
```suggestion
    @test_logs Distributions.AffineDistribution(1.0, 1, Normal())
```
(IMO it's not particularly useful to check the type). Here `AffineDistribution` has to be qualified since it is not exported.